### PR TITLE
Store metadata in well-defined properties

### PR DESCRIPTION
**Pull request by revmischa**
Standardizes the storing of metadata on class views, methods, and functions. 
Adds ability to define routes for a class view, a sub-route for a method, and any route for any function.

Still to do:
* Documentation
* Implement method dispatch
* CRUD

```ts
import { HttpMethod } from "@aws-cdk/aws-apigatewayv2";
import { APIGatewayProxyHandlerV2 } from "aws-lambda";
import { BaseModel } from "../database/baseModel";
import { Column, Entity } from "typeorm";
import { APIEvent } from "../api/base";
import { CrudApiView } from "../api/crud/base";
import { CrudApi, Route, SubRoute } from "../registry";

// sample database model
@Entity()
export class Album extends BaseModel {
  @Column({ nullable: true })
  title: string;
}

// sample CRUD view
@CrudApi({
  model: Album,
  path: "/album",
  memorySize: 512,
  environment: {
    LOG_LEVEL: "DEBUG",
  },
})
export class AlbumCrudApi extends CrudApiView {
  // custom endpoint in the view
  @SubRoute({
    path: "/test",
    methods: [HttpMethod.PATCH],
    environment: {
      LOG_LEVEL: "DEBUG",
    },
  })
  async test() {
    return "Testerino";
  }

  // override post handler
  post: APIGatewayProxyHandlerV2 = async () => "Posterino";
}

// a simple standalone function with a route attached
export async function blargleFunc(event: APIEvent) {
  return JSON.stringify({
    message: "function route",
    rawQueryString: event.rawQueryString,
  });
}
Route({
  path: "/blargle",
  methods: [HttpMethod.PUT],
  memorySize: 384,
  environment: {
    LOG_LEVEL: "WARN",
  },
})(blargleFunc);

// alternate, uglier way of writing the same thing
const blargleFuncInner = Route({
  path: "/blargleInner",
  methods: [HttpMethod.PUT],
  memorySize: 384,
  environment: {
    LOG_LEVEL: "WARN",
  },
})(async function blargleFuncInner(event: APIEvent) {
  return `cookies: ${event.cookies}`;
});
export { blargleFuncInner };
```

Screen Shot 2021-04-24 at 1 20 22 PM

After this PR, development of the `@jetkit/cdk` module will be moved to https://github.com/jetbridge/jetkit-cdk

### CODE REVIEW

File: packages/jetkit-cdk/src/test/sampleApp.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+// sample CRUD view
+@CrudApi({
+  model: Album,
+  path: "/album",
+  memorySize: 512,
+  environment: {
+    LOG_LEVEL: "DEBUG",
+  },
```
Comment (Dizzzmas) [inline packages/jetkit-cdk/src/test/sampleApp.ts:23]: Will this set the log level only for Lambdas generated by this decorator?

Would be cool to have a config in 1 place and just use it everywhere like we do with Python.

Maybe with this https://www.npmjs.com/package/convict ?

**Review by Dizzzmas — APPROVED** [review]
lgtm 🔬 🔬  

Hope soon we'll have `determine`- instead of `guessEntrypoint`

File: packages/jetkit-cdk/src/test/sampleApp.ts (hunk 1 above)
Comment (revmischa) [reply packages/jetkit-cdk/src/test/sampleApp.ts:23]: > Will this set the log level only for Lambdas generated by this decorator?

Yeah, it's just a demo. We'll need some way to set top-level config defaults


> Would be cool to have a config in 1 place and just use it everywhere like we do with Python.
> Maybe with this https://www.npmjs.com/package/convict ?

Sounds great to me! I was imagining a very very thin "app" that jetkit/cdk provides that has config, logging, maybe some plugin architecture but not much else.

**Comment by revmischa** [comment]
Turns out `guessEntrypoint` doesn't work so well with minification. The only way I'm aware of right now to deal with it is to have every decorator call add `entry: __filename` which works but is annoying to always have to provide. So I'm hoping there is some way around that

**Pull request merged**